citesphere/src/test/java/edu/asu/diging/citesphere/core/util/model/impl/CitationHelperTest.java

```diff
@@ -0,0 +1,350 @@
+        conceptTagsSet.add(tag1);
+        updatedCitation.setConceptTags(conceptTagsSet);
+    }
+
+    @Test
+    public void test_updateCitation() {
+        ICitation citation;
+        citation = new Citation();
+        citation.setTitle("title");
+        citation.setShortTitle("shortTitle");
+        citation.setDateFreetext("2019");
+        citation.setVolume("1.0");
+        ICitationConcept concept = new CitationConcept();
+        concept.setId("CON1");
+        concept.setOwner(user);
+        concept.setUri("www.google.com");
+        IConceptType type = new ConceptType();
+        type.setId("CTY1");
+        type.setOwner(user);
+        type.setName("forum");
+        type.setUri("www.test.com");
+        Mockito.when(conceptManager.save(any(ICitationConcept.class))).thenReturn(concept);
+        Mockito.when(typeManager.save(any(IConceptType.class))).thenReturn(type);
+        helperToTest.updateCitation(citation, form, user);
+        assertTrue(equalsCitation(citation, updatedCitation));
+    }
+
+    @Test
+    public void test_updateCitation_getByURIAndOwner() {
+        ICitation citation;
+        citation = new Citation();
+        citation.setTitle("title");
+        citation.setShortTitle("shortTitle");
+        citation.setDateFreetext("2019");
+        citation.setVolume("1.0");
+        ICitationConcept concept = new CitationConcept();
+        concept.setId("CON1");
+        concept.setOwner(user);
+        concept.setUri("www.google.com");
+        IConceptType type = new ConceptType();
+        type.setId("CTY1");
+        type.setOwner(user);
+        type.setName("forum");
+        type.setUri("www.test.com");
+        
+        Mockito.when(conceptManager.getByUriAndOwner(Mockito.argThat(new URIMatcher()), any(IUser.class))).thenReturn(concept);
+
+        
+        Mockito.when(typeManager.getByUriAndOwner(Mockito.argThat(new URIMatcher()), any(IUser.class))).thenReturn(type);
+        helperToTest.updateCitation(citation, form, user);
+        assertTrue(equalsCitation(citation, updatedCitation));
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void test_updateCitation_getByURIAndOwner_Invalid_URI() {
+        ICitation citation;
+        citation = new Citation();
+        citation.setTitle("title");
+        citation.setShortTitle("shortTitle");
+        citation.setDateFreetext("2019");
+        citation.setVolume("1.0");
+        ICitationConcept concept = new CitationConcept();
+        concept.setId("CON1");
+        concept.setOwner(user);
+        concept.setUri("www.google.com");
+        IConceptType type = new ConceptType();
+        type.setId("CTY1");
+        type.setOwner(user);
+        type.setName("forum");
+        type.setUri("www.test.com");
+        form.getConceptTags().iterator().next().setConceptUri("google");
+        
+        Mockito.when(conceptManager.getByUriAndOwner(Mockito.argThat(new URIMatcher()), any(IUser.class))).thenReturn(concept);
```

Comment:
you only want concept be returned, when the user and the concept uri is given, right? Not for any uri and any user.